client/app/services/query.js

```diff
@@ -35,13 +35,14 @@ class Parameters {
     this.initFromQueryString(queryString);
   }
 
-  parseQuery() {
+  parseQuery(queryText = null) {
```

Comment:
```suggestion
  parseQuery(queryText = this.query.query) {
```

(and line 40 could be removed)

Reply:
I thought this wasn't possible  😲😎

Reply:
babel compiles it to something like: 
```javascript
queryText = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : this.query.query
```

so no magic :slightly_smiling_face: